addons/website/controllers/main.py

```diff
@@ -284,8 +284,7 @@ def get_dynamic_snippet_filters(self):
         return dynamic_filter
 
     @http.route('/website/snippet/filter_templates', type='json', auth='public', website=True)
-    def get_dynamic_snippet_templates(self, filter_id=False):
-        # todo: if filter_id.model -> filter template
+    def get_dynamic_snippet_templates(self, mode=False):
```

Comment:
mode = filter_name

and if filter_name: .filter('_' + filter_name + '_')
And remove override from eshop ?

Reply:
I still need an override of the filtering mechanism though: it is a filter that removes the non-requested templates.
If no filter_name specified or not product: get rid of product-specific templates.
Later: if no filter_name specified or not blog: get rig of blog-specific templates... etc...

Reply:
ou on peut partir du principe que si pas de filte rname ou les affiches tous... 
Sachant que chaque override du codebase 'blog, product, event' ajoutera son filter name ...
Qu'en penses-tu pour eviter de faire "usine a gaz"...

Donc si tu utilise le dynamic filter du debug mode (very advanced qui va peut être meme ne plus être droppable un jour vu que qqun a l'air d'y tenir, a ce qu'on ne puisse plus faire de custo sur le saas facilement...), a toi de choisir dans la liste de 'tous les templates" disponible le bon...

Reply:
Ok, mais alors il faut rendre suffisament robuste les templates spécifiques pour qu'ils ne crashent pas quand on leur donne des données inadéquates

Reply:
simple .get('') au lieu de [''] devrait suffire, mais pas "critique" imo.

Reply:
Ok. Just one detail: I have put the filtering itself inside a dedicated method so that its documentation describes what it is about.